python/src/spark_rapids_ml/umap.py

```diff
@@ -0,0 +1,280 @@
+from .utils import _ArrayOrder, _concat_and_free
+
+
+class UMAPClass(_CumlClass):
+    @classmethod
+    def _param_mapping(cls) -> Dict[str, Optional[str]]:
+        return {}
+
+    def _get_cuml_params_default(self) -> Dict[str, Any]:
+        return {
+            "n_neighbors": 15,
+            "n_components": 2,
+            "metric": "euclidean",
+            "n_epochs": None,
+            "learning_rate": 1.0,
+            "init": "spectral",
+            "min_dist": 0.1,
+            "spread": 1.0,
+            "set_op_mix_ratio": 1.0,
+            "local_connectivity": 1.0,
+            "repulsion_strength": 1.0,
+            "negative_sample_rate": 5,
+            "transform_queue_size": 4.0,
+            "a": None,
+            "b": None,
+            "hash_input": False,
+            "random_state": None,
+            "callback": None,
+            "verbose": False,
+            "output_type": None,
+            "handle": None,
+        }
+
+
+class _UMAPCumlParams(_CumlParams, HasFeaturesCol, HasFeaturesCols, HasLabelCol):
+    def __init__(self) -> None:
+        super().__init__()
+        self._setDefault()
+
+    def getFeaturesCol(self) -> Union[str, List[str]]:  # type: ignore
+        """
+        Gets the value of :py:attr:`featuresCol` or :py:attr:`featuresCols`
+        """
+        if self.isDefined(self.featuresCols):
+            return self.getFeaturesCols()
+        elif self.isDefined(self.featuresCol):
+            return self.getOrDefault("featuresCol")
+        else:
+            raise RuntimeError("featuresCol is not set")
+
+    def setFeaturesCol(self: P, value: Union[str, List[str]]) -> P:
+        """
+        Sets the value of :py:attr:`featuresCol` or :py:attr:`featuresCols`. Used when input vectors are stored in a single column.
+        """
+        if isinstance(value, str):
+            self.set_params(featuresCol=value)
+        else:
+            self.set_params(featuresCols=value)
+        return self
+
+    def setFeaturesCols(self: P, value: List[str]) -> P:
+        """
+        Sets the value of :py:attr:`featuresCols`. Used when input vectors are stored as multiple feature columns.
+        """
+        return self.set_params(featuresCols=value)
+
+    def setLabelCol(self: P, value: str) -> P:
+        """
+        Sets the value of :py:attr:`labelCol`.
+        """
+        return self.set_params(labelCol=value)
+
+
+class UMAP(UMAPClass, _CumlEstimatorSupervised, _UMAPCumlParams):
+
+    """
+    >>> from spark_rapids_ml.umap import UMAP
+    >>> X, _ = (1000, 10, centers=42, cluster_std=0.1,
+                dtype=np.float32, random_state=10)
+    >>> df = spark.createDataFrame(X, ["features"])
+    >>> local_model = UMAP().setFeaturesCol("features")
+    >>> distributed_model = umap.fit(df)
+    >>> embeddings = distributed_model.transform(df)
+    >>> embeddings.show()
+    """
+
+    def __init__(self, sample_fraction: float = 1.0, **kwargs: Any) -> None:
```

Comment:
please expand the kwargs to the specific parameters, so we can have nice documentaion

Reply:
I mean can we add the specific function parameters into `__init__` function

``` python
def __init__(self, n_neighbors: int = 25, n_components: int = 2, ...)
```